Jenkinsfile_promote_salt_packages

```diff
@@ -9,49 +9,83 @@ properties([
+        string(defaultValue: '3006.0', description: 'Salt version to promote from products:testing to products.', name: 'salt_version')
+        string(defaultValue: '', description: 'SUSE Manager maintenance update version that this Salt update is released with, e.g. 4.3.7', name: 'mu_version')
+        booleanParam(name: 'recreate_salt_mu_branches', defaultValue: false, description: 'Advanced: Uncheck this if you want to recreate the already existing MU branches at "openSUSE/salt" repository')
     }
 
     agent { label 'manager-jenkins-node' }
 
     stages {
-        stage('Initial Checks') {
+        stage('Create temporary environment for the pipeline') {
             steps {
-                echo "Check that 'openSUSE/MU/${mu_version}' branch exists at https://github.com/openSUSE/salt"
-                sh "curl -I --fail https://codeload.github.com/openSUSE/salt/tar.gz/openSUSE/MU/${mu_version}"
+                sh "mkdir /tmp/salt-promote-pipeline-env || true"
+            }
+        }
 
-                echo "Check that 'MU/${mu_version}' branch exists at https://github.com/openSUSE/salt-packaging"
-                sh "curl -I --fail https://codeload.github.com/openSUSE/salt-packaging/tar.gz/MU/${mu_version}"
+        stage('Initial check and MU branches preparations') {
+	    steps {
+                script {
+                    echo "Check that 'openSUSE/MU/${mu_version}' branch exists at https://github.com/openSUSE/salt"
+                    salt_mu_branch = sh(script: "curl -I --fail https://codeload.github.com/openSUSE/salt/tar.gz/openSUSE/MU/${mu_version}", returnStatus: true)
+                    echo "Check that 'MU/${mu_version}' branch exists at https://github.com/openSUSE/salt-packaging"
+                    salt_packaging_mu_branch = sh(script: "curl -I --fail https://codeload.github.com/openSUSE/salt-packaging/tar.gz/MU/${mu_version}", returnStatus: true)
 
-                echo "Check that 'products:testing' and 'products:testing:debian' are not set to MU branches"
-                sh "! curl https://build.opensuse.org/package/view_file/systemsmanagement:saltstack:products:testing/salt/_service?expand=1 | grep MU/${mu_version}"
-                sh "! curl https://build.opensuse.org/package/view_file/systemsmanagement:saltstack:products:testing:debian/salt/_service?expand=1 | grep MU/${mu_version}"
+                    dir('/tmp/salt-promote-pipeline-env') {
+                        if (salt_mu_branch != 0 || salt_packaging_mu_branch != 0) {
+                            echo "MU branches do not exist. Creating them"
+                            sh "git clone --branch openSUSE/release/${salt_version} --depth 1 https://github.com/openSUSE/salt"
+                            dir('/tmp/salt-promote-pipeline-env/salt') {
+                                sh "git checkout -b openSUSE/MU/${mu_version}"
```

Comment:
Oh, nice. Thanks for the suggestion!

I'm used to use `git checkout -b` to create a new branch based on the current HEAD and switch to it. You mentioned `git-checkout` somehow does this differently, how is this?

Out of curiosity: is there any particular reason to use `switch --create` in favor of `checkout -b` besides of cosmetical reasons? As `git-switch` is a more explicit/concrete command while `checkout` is actually used for many different things.

I'm perfectly fine using `git-switch` instead, which as mentioned seems more concrete command, but I wonder if there is some more reasons. TIA!

Reply:
> I'm used to use `git checkout -b` to create a new branch based on the current HEAD and switch to it. You mentioned `git-checkout` somehow does this differently, how is this?
> 

I can't quite follow, can you rephrase that question? Are you confused because I used `git-checkout` instead of `git checkout`? That's just because many git commands are implemented in separated programs, e.g. `git foo` loads `git-foo`.

> Out of curiosity: is there any particular reason to use `switch --create` in favor of `checkout -b` besides of cosmetical reasons? As `git-switch` is a more explicit/concrete command while `checkout` is actually used for many different things.
> 

Not really, `git-switch` was only added to improve the UX with clearer commands (along with `git-restore`) in 2.23.0 ([Changelog](https://github.com/git/git/blob/master/Documentation/RelNotes/2.23.0.txt#L61))

Reply:
> [...] I was just wondering why you mentioned that:
> 
> _git-switch has a long option for "create branch and switch to it", while **git-checkout does not.**_
> 
> My understanding is that, here in this context, the behavior of both `git switch --create` and `git checkout -b` executions would be the same: create a new branch and swich to it.
> 
> So just wondered why you mentioned that `git checkout -b` wouldn't be creating new branch and switch to it.
> 

That's not what I meant. The emphasis should be on the **long option** part of the sentence. `git-checkout` only has a short option (`-b`) while `git-switch` has a short (`-c`) and a long (`--create`) option.

Reply:
Got it! Sorry for the misunderstanding!